cirq-core/cirq/circuits/circuit.py

```diff
@@ -1173,7 +1173,15 @@ def to_text_diagram_drawer(
         Returns:
             The TextDiagramDrawer instance.
         """
-        qubits = ops.QubitOrder.as_qubit_order(qubit_order).order_for(self.all_qubits())
+        qubits: Tuple[Any, ...] = ops.QubitOrder.as_qubit_order(qubit_order).order_for(
+            self.all_qubits()
+        )
+        cbits = tuple(
+            sorted(
+                (key for op in self.all_operations() for key in protocols.control_keys(op)), key=str
+            )
+        )
+        qubits = qubits + cbits
```

Comment:
"qubits" is no longer just qubits - we should use a different variable name here to avoid confusion.

Reply:
I was thinking that might be better to do in a separate PR, as this renaming needs to go pretty deep (https://github.com/daxfohl/Cirq/compare/doublelines...daxfohl:doublelines2?expand=1). That way all the renaming doesn't distract from the logic here. What do you think?

Reply:
SGTM. If possible, that PR should follow closely after this one - though it does impact several files, it looks like it should be a fairly quick review.

Reply:
Added that as a separate PR #4675 (currently draft)